.github/workflows/tests.yml

```diff
@@ -0,0 +1,29 @@
+---
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  tests:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.10"]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Install dependencies
+        run: pip install .[test]
+      - name: Lint
+        run: flake8
+      - name: Test
+        run: pytest --cov=mbtools --cov-report=term
```

Comment:
@rnathuji Can you help me understand what this yml file does? Is this to set up the python environment?

Reply:
Ah -- this YAML file configures a GitHub workflow so we can use GH Actions for CI. You can find some documentation on that [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions). The way this is setup, it will run our linter and unit tests whenever a PR is created against `main` using both `3.8` and `3.10` versions of Python. It's mostly a copy / paste of what's in our [cli repo](https://github.com/openstax/raise-moodlecli/blob/main/.github/workflows/tests.yml).